traitsui/testing/tester/query.py

```diff
@@ -16,6 +16,30 @@
 """
 
 
+class Selected:
+    """ An object representing an interaction to obtain the object which is
+    currently selected.
+
+    Implementations should return the selected object, or None if nothing is
+    selected.
+    """
+    pass
+
+
+class SelectedIndices:
+    """ An object representing an interaction to obtain the indices of objects
+    which are currently selected.
+
+    Implementations should return either a list of indicies of the selected
+    objects, a single index of a lone selected object, or -1 if nothing is
+    selected.
```

Comment:
What's the rationality behind this API? Why not just return `None` if nothing is selected? Also, why not always return a list - which might contain one or more elements depending on the size of the selection.

Reply:
TBH it was because I am currently just mirroring the implementation of `selected_indices` in the `TableEditor` itself. See comment: https://github.com/enthought/traitsui/pull/1707#discussion_r661660617 and code:
https://github.com/enthought/traitsui/blob/762f66b146b676aaa3b7f1dfcc3d65118976822f/traitsui/qt4/table_editor.py#L627-L643

I agree this is confusing though and lazy.  I can at the very least wrap this with a special handler to give a more consistent return type.  The `SelectedIndices` query class has no reason to be at all tied with a specific Editor (in fact it definitely shouldn't be).

Nonetheless I do still have the question of whether or not I should be reimplementing/rewriting the above code in the UITester TableEditor support code as a specifc handler for the `SelectedIndices` query.  As mentioned in my comment I think that is a little unnecessary / we have this strange chase your own tail scenario... 🤔

Reply:
I need to carefully think about the best api for Selected and SelectedIndices before pushing this through 🤔 

It i tricky because with a TableEditor, one could have rows, columns, cells, etc. selected.  The api needs to accommodate this use case but also be general and not overly complex for other Editors to use

Reply:
I am going to update `Selected` so that it also always returns a list (either a singleton list for one selection, list of all objects selected, or an empty list if nothing is selected).
It is also a bit strange with `SelectedText` as in that case it doesn't really make sense to want to make that query if multiple objects are selected (at least I can't imagine a use case).  Hence for that I think I will not update it to enforce that it returns a list.